src/fides/api/ops/api/v1/endpoints/connection_endpoints.py

```diff
@@ -202,34 +211,95 @@ def get_connection_detail(
     failed: List[BulkUpdateFailed] = []
     logger.info("Starting bulk upsert for {} connection configuration(s)", len(configs))
 
     for config in configs:
+        if config.connection_type == "saas":
+            if config.secrets:
+                try:
+                    connection_config_check = get_connection_config_or_error(
+                        db, config.key
+                    )
+                except HTTPException:
+                    # New config so nothing to validate against
+                    connection_config_check = None
+
+                # This is here rather than with the get_connection_config_or_error because
+                # it will also throw an HTTPException if validation fails and we don't want
+                # to catch it in this case.
+                if connection_config_check:
+                    config.secrets = validate_secrets(
+                        db, config.secrets, connection_config_check
+                    )
+                else:
+                    if not config.saas_connection_type:
+                        raise HTTPException(
+                            status_code=HTTP_404_NOT_FOUND,
+                            detail="SAAS config type is missing",
+                        )
+
+                    registry = load_registry(registry_file)
+                    connector_template = registry.get_connector_template(
+                        config.saas_connection_type
+                    )
+                    if not connector_template:
+                        raise HTTPException(
+                            status_code=HTTP_404_NOT_FOUND,
+                            detail=f"SaaS connector type '{config.saas_connection_type}' is not yet available in Fides. For a list of available SaaS connectors, refer to {CONNECTION_TYPES}.",
+                        )
+                    try:
+                        template_values = SaasConnectionTemplateValues(
+                            name=config.name,
+                            key=config.key,
+                            description=config.description,
+                            secrets=config.secrets,
+                            instance_key=config.key,
+                        )
+                        connection_config = (
+                            create_connection_config_from_template_no_save(
+                                db, connector_template, template_values
+                            )
+                        )
+                    except KeyOrNameAlreadyExists as exc:
+                        raise HTTPException(
+                            status_code=HTTP_400_BAD_REQUEST,
+                            detail=exc.args[0],
+                        )
+
+                    connection_config.secrets = validate_secrets(
+                        db, template_values.secrets, connection_config
+                    ).dict()
+                    connection_config.save(db=db)
```

Comment:
I tried this, but it caused circular import errors. I'm sure I could come up with a bigger refactor that would allow breaking the circular imports but it would probably take a lot more changes.

Reply:
ah no, that's fine then!